src/main/java/com/pancakeswap/nft/publish/service/MoboxTokenService.java

```diff
@@ -26,25 +28,25 @@ public class MoboxTokenService {
     private final AttributeRepository attributeRepository;
     private final TokenRepository tokenRepository;
     private final static String LEVEL_ATTRIBUTE = "lv";
+    private final static int PAGE_SIZE = 1000;
 
     public void updateLevels(Collection collection) {
         String collectionAddress = collection.getAddress();
-
         String collectionId = collection.getId();
-        List<Token> tokens = tokenRepository.findAllByParentCollection(new ObjectId(collectionId));
 
-        tokens.forEach(token -> {
-            List<String> attributesId = token.getAttributes().stream()
-                    .map(ObjectId::toString)
-                    .collect(Collectors.toList());
+        PageRequest pageRequest = PageRequest.ofSize(PAGE_SIZE);
+        Page<Token> tokens = tokenRepository.findAllByParentCollection(new ObjectId(collectionId), pageRequest);
+
+        tokens.stream().forEach(token -> {
```

Comment:
Here you fetch only one page with a size of 1000. What if tokens will be more than 1000?

Reply:
Thanks for spotting this. Let me please fix this